Packs/CommonScripts/Scripts/GetEndpointData/GetEndpointData.py

```diff
@@ -0,0 +1,719 @@
+                output_keys=["Endpoint"],
+                args_mapping={},
+                output_mapping={},
+                post_processing=cylance_filtering
+            )
+        ]
+
+        list_args_commands = [
+            MappedCommand(
+                brand='Cortex XDR - IR',
+                name='xdr-get-endpoints',
+                output_keys=['Endpoint', 'Account'],
+                args_mapping={'endpoint_id_list': 'agent_id', 'ip_list': 'agent_ip', 'hostname': 'agent_hostname'},
+                output_mapping={}
+            ),
+            MappedCommand(
+                brand='Cortex Core - IR',
+                name='core-list-risky-hosts',
+                output_keys=["Endpoint"],
+                args_mapping={'host_id': 'agent_id'},
+                output_mapping={'id': 'ID'},
+
+            ),
+            MappedCommand(
+                brand='CrowdstrikeFalcon',
+                name='cs-falcon-search-device',
+                output_keys=["Endpoint"],
+                args_mapping={'ids': 'agent_id', 'hostname': 'agent_hostname'},
+                output_mapping={},
+            )
+        ]
+
+        # Run a loop for commands that do not take an array as an input
+        for agent_id, agent_ip, agent_hostname in zip_longest(agent_ids, agent_ips, agent_hostnames, fillvalue=""):
+            single_endpoint_outputs = []
+            single_endpoint_readable_outputs = []
+
+            for command in single_args_command:
+                readable_outputs, endpoint_output = command_runner.run_command(
+                    command,
+                    {
+                        'agent_id': agent_id,
+                        'agent_ip': agent_ip,
+                        'agent_hostname': agent_hostname
+                    }
+                )
+
+                if endpoint_output:
+                    single_endpoint_outputs.append(endpoint_output)
+
+                else:
+                    demisto.debug(f'endpoint not found {agent_ids=} {agent_ips=} {agent_hostnames=}')
+                    endpoints_not_found_list.append({
+                        'Key': agent_id or agent_ip or agent_hostname,
+                        'Source': command.brand
+                    })
+                single_endpoint_readable_outputs.extend(readable_outputs)
+
+            if verbose:
+                command_results_list.extend(single_endpoint_readable_outputs)
+
+            for index in range(max(map(len, single_endpoint_outputs), default=0)):
+                unmerged_endpoints = [safe_list_get(lst, index, {}) for lst in single_endpoint_outputs]
+                demisto.debug(f'merging endoints {unmerged_endpoints=}, {index=}')
+                merged_endpoint = merge_endpoints(unmerged_endpoints) if unmerged_endpoints else None
+                if merged_endpoint:
+                    demisto.debug(f'appending {merged_endpoint=}')
+                    endpoint_outputs_list.append(merged_endpoint)
```

Comment:
This code seems to be duplicated in the main, please think of a way to combine them.
Also, the code is not clear so let's think of a solution to make it easier to read.